src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -62,85 +119,108 @@ int FilterParserThread::parseDATFilterFile()
         return ruleCount;
     }
 
-    unsigned int nbLine = 0;
-    while (!file.atEnd() && !m_abort) {
-        ++nbLine;
-        QByteArray line = file.readLine();
-        // Ignoring empty lines
-        line = line.trimmed();
-        if (line.isEmpty()) continue;
-        // Ignoring commented lines
-        if (line.startsWith('#') || line.startsWith("//")) continue;
+    static const int bufferSize = 2 * 1024 * 1024; // 2 MiB
+    char *buffer = new char[bufferSize];
+    qint64 bytesRead = 0;
+    int offset = 0;
+    int start = 0;
+    int endOfLine = -1;
+    int nbLine = 0;
+
+    while (true) {
+        memset(buffer + offset, 0, bufferSize - offset);
+        bytesRead = file.read(buffer + offset, bufferSize - offset);
+        if (bytesRead < 1) break;
+        int dataSize = bytesRead + offset;
+
```

Comment:
I would move this empty line one line up (to separate break statement from the rest).